packages/remix-react/components.tsx

```diff
@@ -1436,7 +1436,9 @@ export type FetcherWithComponents<TData> = Fetcher<TData> & {
  *
  * @see https://remix.run/api/remix#usefetcher
  */
-export function useFetcher<TData = any>(): FetcherWithComponents<TData> {
+export function useFetcher<TData = any>(): FetcherWithComponents<
+  SerializeFrom<TData>
```

Comment:
Wouldn't `SerializeFrom`, when applied to itself, still produce the correct result?  e.g. `SerializeFrom<SerializeFrom<typeof action>>` I believe will still resolve to the same type, because the inner `SerializeFrom<typeof action>` will produce a serialized object, e.g. `{ foo: string }`, and then `SerializeFrom<{ foo: string }>` will still extend `{ foo: string }`, which is the correct type of the fetcher data.   But maybe there's edge cases here I'm not considering?

If that's incorrect and `SerializeFrom` doesn't support nesting, would it be worth adding a type for backwards compatibility to unwrap the type if the user is using it in the old way?  e.g.:
```ts
// If the type already extends SerializeFrom, unwrap the inner type before applying SerializeFrom again
type UnwrapSerializeFrom<T> = T extends SerializeFrom<infer U> ? U : T;


export function useFetcher<TData = any>(): FetcherWithComponents<
  SerializeFrom<UnwrapSerializeFrom<TData>>
>  { 
// ...
}
```

Reply:
You're correct, I thought it couldn't, but `SerializeFrom` can be nested as much as possible